### PR TITLE
Unit tests and common base class for IHostEnvironment Layout Renderers

### PR DESCRIPTION
**Pull request by bakgerman**
Derived a common base class to allow for unit testability of the layout renderers using IHost(ing)Environment and then added the unit tests to those layout renderers.

### CODE REVIEW

File: src/Shared/LayoutRenderers/AspNetHostEnvironmentLayoutRendererBase.cs [hunk 1]
```diff
@@ -0,0 +1,66 @@
...
+        }
+#else
+        /// <summary>
+        /// Provides access to the current IHostEnvironment
+        /// </summary>
+        /// <returns>IHostEnvironment or <c>null</c></returns>
+        [NLogConfigurationIgnoreProperty]
+        public IHostEnvironment HostEnvironment
+        {
+            get => _hostEnvironment ?? (_hostEnvironment = new HostEnvironment());
```
Comment (snakefoot) [inline src/Shared/LayoutRenderers/AspNetHostEnvironmentLayoutRendererBase.cs:54]: Think this is weird.

File: src/Shared/LayoutRenderers/IISInstanceNameLayoutRenderer.cs [hunk 2]
```diff
@@ -29,31 +28,16 @@ namespace NLog.Web.LayoutRenderers
     [LayoutRenderer("iis-site-name")]
     // ReSharper disable once InconsistentNaming
     [ThreadAgnostic]
-    public class IISInstanceNameLayoutRenderer : LayoutRenderer
+    public class IISInstanceNameLayoutRenderer : AspNetHostEnvironmentLayoutRendererBase
```
Comment (snakefoot) [inline src/Shared/LayoutRenderers/IISInstanceNameLayoutRenderer.cs:31]: Think the complexity of "inheriting" the ability to lookup HostEnvironment outweights the code-use.

File: src/Shared/LayoutRenderers/AspNetHostEnvironmentLayoutRendererBase.cs (hunk 1 above)
Comment (bakgerman) [reply src/Shared/LayoutRenderers/AspNetHostEnvironmentLayoutRendererBase.cs]: Simplified

File: src/Shared/LayoutRenderers/IISInstanceNameLayoutRenderer.cs (hunk 2 above)
Comment (bakgerman) [reply src/Shared/LayoutRenderers/IISInstanceNameLayoutRenderer.cs:31]: OK, that can be adjusted

Comment (bakgerman) [reply src/Shared/LayoutRenderers/IISInstanceNameLayoutRenderer.cs]: removed common base class

File: src/Shared/LayoutRenderers/AspNetAppBasePathLayoutRenderer.cs [hunk 3]
```diff
@@ -23,80 +23,87 @@ namespace NLog.Web.LayoutRenderers
...
-            builder.Append(AppBasePath);
+            return ServiceLocator.ResolveService<IHostEnvironment>(serviceProvider, loggingConfiguration);
         }
+#else
+        /// <summary>
+        /// Provides access to the current IHostEnvironment
+        /// </summary>
+        /// <returns>IHostEnvironment or <c>null</c></returns>
+        [NLogConfigurationIgnoreProperty]
+        public IHostEnvironment HostEnvironment { get; set; } = new HostEnvironment();
```
Comment (snakefoot) [inline src/Shared/LayoutRenderers/AspNetAppBasePathLayoutRenderer.cs:60]: Can this become `internal` and changed so `HostEnvironment`-class has a static-property that points to a `Default`-instance:

`public IHostEnvironment HostEnvironment { get; set; } = HostEnvironment.Default;`

File: src/Shared/LayoutRenderers/AspNetAppBasePathLayoutRenderer.cs [hunk 4]
```diff
@@ -23,80 +23,87 @@ namespace NLog.Web.LayoutRenderers
...
-#else
-        private string AppBasePath => _appBasePath ?? ResolveAppBasePath(HostingEnvironment.MapPath("~"), out _appBasePath);
-        private static string _appBasePath;
-#endif
+        /// <summary>
+        /// Provides access to the current IHostEnvironment
+        /// </summary>
+        /// <returns>IHostEnvironment or <c>null</c></returns>
+        [NLogConfigurationIgnoreProperty]
+        public IHostEnvironment HostEnvironment
```
Comment (snakefoot) [inline src/Shared/LayoutRenderers/AspNetAppBasePathLayoutRenderer.cs:44]: Can this become internal ?

File: src/Shared/LayoutRenderers/AspNetAppBasePathLayoutRenderer.cs (hunk 3 above)
Comment (bakgerman) [reply src/Shared/LayoutRenderers/AspNetAppBasePathLayoutRenderer.cs]: Complete. forgot I did the request body middleware options the same way

File: src/Shared/LayoutRenderers/AspNetAppBasePathLayoutRenderer.cs (hunk 4 above)
Comment (bakgerman) [reply src/Shared/LayoutRenderers/AspNetAppBasePathLayoutRenderer.cs:44]: Complete

File: src/NLog.Web.AspNetCore/LayoutRenderers/AspNetEnvironmentLayoutRenderer.cs [hunk 5]
```diff
@@ -20,21 +20,31 @@ namespace NLog.Web.LayoutRenderers
...
+        /// </summary>
         private IHostEnvironment _hostEnvironment;
 
-        private IHostEnvironment HostEnvironment => _hostEnvironment ?? (_hostEnvironment = ServiceLocator.ResolveService<IHostEnvironment>(ResolveService<IServiceProvider>(), LoggingConfiguration));
+        /// <summary>
+        /// Provides access to the current IHostEnvironment
+        /// </summary>
+        /// <returns>IHostEnvironment or <c>null</c></returns>
+        [NLogConfigurationIgnoreProperty]
+        public IHostEnvironment HostEnvironment
```
Comment (snakefoot) [inline src/NLog.Web.AspNetCore/LayoutRenderers/AspNetEnvironmentLayoutRenderer.cs:33]: Can this become internal ?

File: src/Shared/LayoutRenderers/AspNetAppBasePathLayoutRenderer.cs (hunk 4 above)
Comment (snakefoot) [inline src/Shared/LayoutRenderers/AspNetAppBasePathLayoutRenderer.cs:44]: Can this become internal ?

File: src/Shared/LayoutRenderers/IISInstanceNameLayoutRenderer.cs [hunk 6]
```diff
@@ -32,28 +33,42 @@ namespace NLog.Web.LayoutRenderers
...
         private IHostEnvironment _hostEnvironment;
 
-        private IHostEnvironment HostEnvironment => _hostEnvironment ?? (_hostEnvironment = ServiceLocator.ResolveService<IHostEnvironment>(ResolveService<IServiceProvider>(), LoggingConfiguration));
-#endif
+        /// <summary>
+        /// Provides access to the current IHostEnvironment
+        /// </summary>
+        /// <returns>IHostEnvironment or <c>null</c></returns>
+        [NLogConfigurationIgnoreProperty]
+        public IHostEnvironment HostEnvironment
```
Comment (snakefoot) [inline src/Shared/LayoutRenderers/IISInstanceNameLayoutRenderer.cs:46]: Can this become internal  (And remove `NLogConfigurationIgnoreProperty`)

File: src/Shared/LayoutRenderers/AspNetAppBasePathLayoutRenderer.cs [hunk 7]
```diff
@@ -23,80 +23,87 @@ namespace NLog.Web.LayoutRenderers
...
+        /// <summary>
+        /// Provides access to the current IHostEnvironment
+        /// </summary>
+        /// <returns>IHostEnvironment or <c>null</c></returns>
+        [NLogConfigurationIgnoreProperty]
+        internal IHostEnvironment HostEnvironment { get; set; } = Internal.HostEnvironment.Default;
+#endif
 
         /// <inheritdoc />
-        protected override void CloseLayoutRenderer()
```
Comment (snakefoot) [inline src/Shared/LayoutRenderers/AspNetAppBasePathLayoutRenderer.cs:51]: Can the clear be restored again on close ?

File: src/Shared/LayoutRenderers/IISInstanceNameLayoutRenderer.cs [hunk 8]
```diff
@@ -32,28 +33,42 @@ namespace NLog.Web.LayoutRenderers
...
             builder.Append(HostEnvironment?.ApplicationName);
 #else
-            builder.Append(HostingEnvironment.SiteName);
+            builder.Append(HostEnvironment?.SiteName);
 #endif
         }
-
-#if ASP_NET_CORE
-        /// <inheritdoc />
-        protected override void CloseLayoutRenderer()
```
Comment (snakefoot) [inline src/Shared/LayoutRenderers/IISInstanceNameLayoutRenderer.cs:52]: Can the clear be restored again on close ?

File: src/NLog.Web.AspNetCore/LayoutRenderers/AspNetEnvironmentLayoutRenderer.cs [hunk 9]
```diff
@@ -20,21 +20,31 @@ namespace NLog.Web.LayoutRenderers
...
-        /// <inheritdoc />
-        protected override void Append(StringBuilder builder, LogEventInfo logEvent)
+        internal static IHostEnvironment RetrieveHostEnvironment(IServiceProvider serviceProvider, LoggingConfiguration loggingConfiguration)
         {
-            builder.Append(HostEnvironment?.EnvironmentName);
+            return ServiceLocator.ResolveService<IHostEnvironment>(serviceProvider, loggingConfiguration);
         }
 
         /// <inheritdoc />
-        protected override void CloseLayoutRenderer()
```
Comment (snakefoot) [inline src/NLog.Web.AspNetCore/LayoutRenderers/AspNetEnvironmentLayoutRenderer.cs:34]: Can the clear be restored again on close ?

**Comment by bakgerman** [comment]
Apologies for the silly mistakes, trying to contribute while at work was maybe not the best idea.  Should be all fixed now.

**Comment by snakefoot** [comment]
No problem. Can `NLogConfigurationIgnoreProperty`-attributes also be removed ?

**Comment by bakgerman** [comment]
removed as asked.  made sure to remove from both conditional compiles this time.

File: src/NLog.Web/IHostEnvironment.cs [hunk 10]
```diff
@@ -0,0 +1,22 @@
...
+using System.Collections.Generic;
+using System.Linq;
+using System.Text;
+
+namespace NLog.Web
+{
+    /// <summary>
+    /// Interface to allow unit testing of System.Web.Hosting.HostingEnvironment based layout renderers
+    /// </summary>
+    public interface IHostEnvironment
```
Comment (snakefoot) [inline src/NLog.Web/IHostEnvironment.cs:11]: Can this become internal, and be moved into NLog.Web.Internal-namespace

**Comment by bakgerman** [comment]
last change caused a unit test failure due to use of internal interface in unit test assembly.  Fixing...

**Comment by bakgerman** [comment]
true, let me make FakeHostEnvironment class in unit test assembly

**Comment by snakefoot** [comment]
I fine with using NSubstitute for AspNetCore. It was only the one for NLog. Web.Internal that could be done manual. Maybe rename the NLog.Web.Internal.IHostEnvironment to something else to avoid confusion in the future.



Sent from my Galaxy



-------- Original message --------
From: Burak Akgerman ***@***.***>
Date: 20/07/2022 22:34 (GMT+01:00)
To: "NLog/NLog.Web" ***@***.***>
Cc: Rolf Kristensen ***@***.***>, Comment ***@***.***>
Subject: Re: [NLog/NLog.Web] Unit tests and common base class for IHostEnvironment Layout Renderers (PR #832)


true, let me make FakeHostEnvironment class in unit test assembly

File: tests/NLog.Web.Tests/FakeHostEnvironment.cs [hunk 11]
```diff
@@ -0,0 +1,88 @@
...
+#endif
+
+namespace NLog.Web.Tests
+{
+    /// <summary>
+    /// Faked implementation of IHostEnvironment designed for unit testing.
+    /// </summary>
+    public class FakeHostEnvironment : IHostEnvironment
+    {
+#if ASP_NET_CORE
```
Comment (snakefoot) [inline tests/NLog.Web.Tests/FakeHostEnvironment.cs:22]: Can the the mock be moved into the NLog.Web-project alone, and be stripped from ASP_NET_CORE-details ?

File: src/NLog.Web/Properties/AssemblyInfo.cs [hunk 12]
```diff
@@ -9,3 +9,4 @@
 // The following GUID is for the ID of the typelib if this project is exposed to COM
 [assembly: Guid("74d5915b-bea9-404c-b4d0-b663164def37")]
 [assembly: InternalsVisibleTo("NLog.Web.Tests,PublicKey=0024000004800000940000000602000000240000525341310004000001000100772391e63c104728adcf18e2390474262559fa7f34a4215848f43288cde875dcc92a06222e9be0592b211ff74adbb5d21a7aab5522b540b1735f2f03279221056fedbe7e534073dabee9db48f8ecebcf1dc98a95576e45cbeff5fe7c4842859451ab2dae7a8370f1b2f7a529d2ca210e3e844d973523d73d193df6c17f1314a6")]
+[assembly: InternalsVisibleTo("DynamicProxyGenAssembly2, PublicKey=0024000004800000940000000602000000240000525341310004000001000100c547cac37abd99c8db225ef2f6c8a3602f3b3606cc9891605d02baa56104f4cfc0734aa39b93bf7852f7d9266654753cc297e7d2edfe0bac1cdcf9f717241550e0a7b191195b7667bb4f64bcb8e2121380fd1d9d46ad2d92d2d15605093924cceaf74c4861eff62abf69b9291ed0a340e113be11e6a7d3113e92484cf7045cc7")]
```
Comment (snakefoot) [inline src/NLog.Web/Properties/AssemblyInfo.cs:12]: Can this be remove again?

File: src/NLog.Web.AspNetCore/Properties/AssemblyInfo.cs [hunk 13]
```diff
@@ -9,3 +9,4 @@
 // The following GUID is for the ID of the typelib if this project is exposed to COM
 [assembly: Guid("74d5915b-bea9-404c-b4d0-b663164def37")]
 [assembly: InternalsVisibleTo("NLog.Web.AspNetCore.Tests,PublicKey=0024000004800000940000000602000000240000525341310004000001000100772391e63c104728adcf18e2390474262559fa7f34a4215848f43288cde875dcc92a06222e9be0592b211ff74adbb5d21a7aab5522b540b1735f2f03279221056fedbe7e534073dabee9db48f8ecebcf1dc98a95576e45cbeff5fe7c4842859451ab2dae7a8370f1b2f7a529d2ca210e3e844d973523d73d193df6c17f1314a6")]
+[assembly: InternalsVisibleTo("DynamicProxyGenAssembly2, PublicKey=0024000004800000940000000602000000240000525341310004000001000100c547cac37abd99c8db225ef2f6c8a3602f3b3606cc9891605d02baa56104f4cfc0734aa39b93bf7852f7d9266654753cc297e7d2edfe0bac1cdcf9f717241550e0a7b191195b7667bb4f64bcb8e2121380fd1d9d46ad2d92d2d15605093924cceaf74c4861eff62abf69b9291ed0a340e113be11e6a7d3113e92484cf7045cc7")]
```
Comment (snakefoot) [inline src/NLog.Web.AspNetCore/Properties/AssemblyInfo.cs:12]: Can this be remove again?

File: src/NLog.Web/Internal/IHostEnvironment.cs [hunk 14]
```diff
@@ -0,0 +1,22 @@
+﻿using System;
+using System.Collections.Generic;
+using System.Linq;
+using System.Text;
+
+namespace NLog.Web
```
Comment (snakefoot) [inline src/NLog.Web/Internal/IHostEnvironment.cs:6]: Can namespace be updated to NLog.Web.Internal

Comment (bakgerman) [reply src/NLog.Web/Internal/IHostEnvironment.cs]: All completed

File: src/Shared/LayoutRenderers/AspNetAppBasePathLayoutRenderer.cs [hunk 15]
```diff
@@ -23,80 +23,89 @@ namespace NLog.Web.LayoutRenderers
...
     public class AspNetAppBasePathLayoutRenderer : LayoutRenderer
     {
 #if ASP_NET_CORE
-        private IHostEnvironment HostEnvironment => _hostEnvironment ?? (_hostEnvironment = ServiceLocator.ResolveService<IHostEnvironment>(ResolveService<IServiceProvider>(), LoggingConfiguration));
         private IHostEnvironment _hostEnvironment;
 
-        private string AppBasePath => _appBasePath ?? ResolveAppBasePath(HostEnvironment?.ContentRootPath, out _appBasePath);
-        private string _appBasePath;
-#else
-        private string AppBasePath => _appBasePath ?? ResolveAppBasePath(HostingEnvironment.MapPath("~"), out _appBasePath);
```
[the remark this answers is not in the document]
Comment (bakgerman) [reply src/Shared/LayoutRenderers/AspNetAppBasePathLayoutRenderer.cs:40]: Fixed.

File: src/Shared/LayoutRenderers/IISInstanceNameLayoutRenderer.cs [hunk 16]
```diff
@@ -34,26 +35,42 @@ public class IISInstanceNameLayoutRenderer : LayoutRenderer
...
+        internal static IHostEnvironment RetrieveHostEnvironment(IServiceProvider serviceProvider, LoggingConfiguration loggingConfiguration)
+        {
+            return ServiceLocator.ResolveService<IHostEnvironment>(serviceProvider, loggingConfiguration);
+        }
+#else
+        /// <summary>
+        /// Provides access to the current IHostEnvironment
+        /// </summary>
+        /// <returns>IHostEnvironment or <c>null</c></returns>
+        internal IHostEnvironment HostEnvironment { get; set; } = Internal.HostEnvironment.Default;
```
Comment (snakefoot) [inline src/Shared/LayoutRenderers/IISInstanceNameLayoutRenderer.cs:57]: Possible to merge the HostEnvironment-properties together, so the only difference is the method used to resolve the `IHostEnvironment`-implementation?

File: src/Shared/LayoutRenderers/AspNetAppBasePathLayoutRenderer.cs [hunk 17]
```diff
@@ -23,80 +23,96 @@ namespace NLog.Web.LayoutRenderers
...
+            return ServiceLocator.ResolveService<IHostEnvironment>(serviceProvider, loggingConfiguration);
+        }
 #else
-        private string AppBasePath => _appBasePath ?? ResolveAppBasePath(HostingEnvironment.MapPath("~"), out _appBasePath);
-        private static string _appBasePath;
+        /// <summary>
+        /// Provides access to the current IHostEnvironment
+        /// </summary>
+        /// <returns>IHostEnvironment or <c>null</c></returns>
+        internal IHostEnvironment HostEnvironment { get; set; } = Internal.HostEnvironment.Default;
```
Comment (snakefoot) [inline src/Shared/LayoutRenderers/AspNetAppBasePathLayoutRenderer.cs:55]: Possible to merge the HostEnvironment-properties together, so the only difference is the method used to resolve the `IHostEnvironment`-implementation?

File: src/NLog.Web.AspNetCore/LayoutRenderers/AspNetEnvironmentLayoutRenderer.cs [hunk 18]
```diff
@@ -22,18 +22,31 @@ public class AspNetEnvironmentLayoutRenderer : LayoutRenderer
...
+        /// Provides access to the current IHostEnvironment
+        /// </summary>
+        /// <returns>IHostEnvironment or <c>null</c></returns>
+        internal IHostEnvironment HostEnvironment
+        {
+            get => _hostEnvironment ?? (_hostEnvironment = RetrieveHostEnvironment(ResolveService<IServiceProvider>(), LoggingConfiguration));
+            set => _hostEnvironment = value;
+        }
+
+        internal static IHostEnvironment RetrieveHostEnvironment(IServiceProvider serviceProvider, LoggingConfiguration loggingConfiguration)
```
Comment (snakefoot) [inline src/NLog.Web.AspNetCore/LayoutRenderers/AspNetEnvironmentLayoutRenderer.cs:35]: Can this become private and non-static, and then remove the input-parameters?

File: src/Shared/LayoutRenderers/AspNetAppBasePathLayoutRenderer.cs [hunk 19]
```diff
@@ -23,80 +23,97 @@ namespace NLog.Web.LayoutRenderers
...
+    /// Rendering Application BasePath. <see cref="IHostEnvironment.MapPath"/>("~")
     /// </summary>
 #endif
     [LayoutRenderer("aspnet-appbasepath")]
     [ThreadAgnostic]
     public class AspNetAppBasePathLayoutRenderer : LayoutRenderer
     {
 #if ASP_NET_CORE
-        private IHostEnvironment HostEnvironment => _hostEnvironment ?? (_hostEnvironment = ServiceLocator.ResolveService<IHostEnvironment>(ResolveService<IServiceProvider>(), LoggingConfiguration));
+        internal static IHostEnvironment RetrieveHostEnvironment(IServiceProvider serviceProvider, LoggingConfiguration loggingConfiguration)
```
Comment (snakefoot) [inline src/Shared/LayoutRenderers/AspNetAppBasePathLayoutRenderer.cs:34]: Can this become private and non-static, and then remove the input-parameters?

File: src/Shared/LayoutRenderers/IISInstanceNameLayoutRenderer.cs [hunk 20]
```diff
@@ -32,28 +33,45 @@ namespace NLog.Web.LayoutRenderers
     public class IISInstanceNameLayoutRenderer : LayoutRenderer
     {
 #if ASP_NET_CORE
+        internal static IHostEnvironment RetrieveHostEnvironment(IServiceProvider serviceProvider, LoggingConfiguration loggingConfiguration)
```
Comment (snakefoot) [inline src/Shared/LayoutRenderers/IISInstanceNameLayoutRenderer.cs:36]: Can this become private and non-static, and then remove the input-parameters?

File: src/NLog.Web.AspNetCore/LayoutRenderers/AspNetEnvironmentLayoutRenderer.cs [hunk 21]
```diff
@@ -22,18 +22,31 @@ public class AspNetEnvironmentLayoutRenderer : LayoutRenderer
...
+        /// <returns>IHostEnvironment or <c>null</c></returns>
+        internal IHostEnvironment HostEnvironment
+        {
+            get => _hostEnvironment ?? (_hostEnvironment = RetrieveHostEnvironment());
+            set => _hostEnvironment = value;
+        }
+
+        private IHostEnvironment RetrieveHostEnvironment()
+        {
+            return ResolveService<IServiceProvider>()?.GetService<IHostEnvironment>();
```
Comment (snakefoot) [inline src/NLog.Web.AspNetCore/LayoutRenderers/AspNetEnvironmentLayoutRenderer.cs:37]: Could this become:
```c#
ServiceLocator.ResolveService<IHostEnvironment>(ResolveService<IServiceProvider>(), LoggingConfiguration)
```

File: src/Shared/LayoutRenderers/AspNetAppBasePathLayoutRenderer.cs [hunk 22]
```diff
@@ -23,80 +22,97 @@ namespace NLog.Web.LayoutRenderers
...
+    /// Rendering Application BasePath. <see cref="IHostEnvironment.MapPath"/>("~")
     /// </summary>
 #endif
     [LayoutRenderer("aspnet-appbasepath")]
     [ThreadAgnostic]
     public class AspNetAppBasePathLayoutRenderer : LayoutRenderer
     {
 #if ASP_NET_CORE
-        private IHostEnvironment HostEnvironment => _hostEnvironment ?? (_hostEnvironment = ServiceLocator.ResolveService<IHostEnvironment>(ResolveService<IServiceProvider>(), LoggingConfiguration));
+        private IHostEnvironment RetrieveHostEnvironment()
```
Comment (snakefoot) [inline src/Shared/LayoutRenderers/AspNetAppBasePathLayoutRenderer.cs:33]: Could this method be moved down below `HostEnvironment`-property, and be changed to this:
```c#
private IHostEnvironment RetrieveHostEnvironment()
{
#if ASP_NET_CORE
    return ServiceLocator.ResolveService<IHostEnvironment>(ResolveService<IServiceProvider>(), LoggingConfiguration);
#else
    return Internal.HostEnvironment.Default
#endif
}
```

File: src/Shared/LayoutRenderers/IISInstanceNameLayoutRenderer.cs [hunk 23]
```diff
@@ -32,28 +33,45 @@ namespace NLog.Web.LayoutRenderers
     public class IISInstanceNameLayoutRenderer : LayoutRenderer
     {
 #if ASP_NET_CORE
+        private IHostEnvironment RetrieveHostEnvironment()
```
Comment (snakefoot) [inline src/Shared/LayoutRenderers/IISInstanceNameLayoutRenderer.cs:36]: Could this method be moved down below `HostEnvironment`-property, and be changed to this:
```c#
private IHostEnvironment RetrieveHostEnvironment()
{
#if ASP_NET_CORE
    return ServiceLocator.ResolveService<IHostEnvironment>(ResolveService<IServiceProvider>(), LoggingConfiguration);
#else
    return Internal.HostEnvironment.Default
#endif
}
```

**Comment by snakefoot** [comment]
@bakgerman Would like to merge this. Any chance to have the last requests resolved?

**Comment by bakgerman** [comment]
> @bakgerman Would like to merge this. Any chance to have the last requests resolved?

Yes I will do that today.

**Comment by snakefoot** [comment]
Thank you again for the nice contribution, and being patient with me.

**Comment by snakefoot** [comment]
Together with #844 now the NLog.Web-repo has crossed the magic 80 pct. code-coverage-barrier. 

@bakgerman Thank you again for all your hard work.

**Pull request merged**